core/src/main/java/de/danoeh/antennapod/core/util/QueueSorter.java

```diff
@@ -86,13 +108,76 @@ public static void sort(final Context context, final Rule rule, final boolean br
             case SMART_SHUFFLE_DESC:
                 permutor = (queue) -> smartShuffle(queue, false);
                 break;
-            default:
         }
 
         if (comparator != null) {
-            DBWriter.sortQueue(comparator, broadcastUpdate);
-        } else if (permutor != null) {
-            DBWriter.reorderQueue(permutor, broadcastUpdate);
+            final Comparator<FeedItem> comparator2 = comparator;
+            permutor = (queue) -> Collections.sort(queue, comparator2);
+        }
+        return permutor;
+    }
+
+    /**
+     * Converts a QueueSortOrder value to its corresponding Rule value.
+     *
+     * @param sortOrder Sort order.
+     * @return Rule value corresponding to the given sort order. <code>null</code> if the sort order is unknown or <code>null</code>.
+     */
+    public static Rule queueSortOrder2Rule(UserPreferences.QueueSortOrder sortOrder) {
```

Comment:
In my opinion, it would be better to have one single sort order enum. Maybe the enum could even be in its own file. That prevents inconsistencies because of the code duplication and does not need converting.

Reply:
I agree, this keeps the code easier. I will change the code as mentioned.